core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -1508,6 +1508,54 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     servers.foreach(assertNoExemptRequestMetric(_))
   }
 
+  @Test
+  def testRateMetricsHaveCumulativeCount() {
+    val numRecords = 1000
+    sendRecords(numRecords)
+
+    val consumer = this.consumers.head
+    consumer.assign(List(tp).asJava)
+    consumer.seek(tp, 0)
+    consumeAndVerifyRecords(consumer, numRecords = numRecords, startingOffset = 0)
+
+    def exists(name: String, metricName: MetricName, allMetricNames: Set[MetricName]): Boolean = {
```

Comment:
Maybe `metricName` should be `rateMetricName` to make it clear why we create a new `MetricName` below.